client/task-list/tasks/payments/wcpay.js

```diff
@@ -38,6 +37,17 @@ class WCPay extends Component {
 				)
 			);
 			markConfigured( 'wcpay' );
+		} else if ( this.props.installStep.isComplete ) {
+			this.connect();
```

Comment:
When using the Back button after redirect, it immediately redirects again. Maybe we should only automatically redirect after actual plugin installation, and keep the explicit action button for other cases?

Reply:
> keep the explicit action button for other cases?

The reason I didn't want an explicit action button is that the copy is hard to write. Before this change it said "Verify details with Stripe", but now it will be "Connect to WP.com". I'm working on removing that page altogether, which is David's ultimate goal.

Reply:
Can you give this another try? I've changed the UX so everything happens in the methods list: https://github.com/woocommerce/woocommerce-admin/pull/4499/commits/fd1cafc9555967aa0f9ef0726d4e46bc41d5fa12

Reply:
Looks like a great change, and works well.

(Haven't looked over the code change in as much detail as I'd like yet, but can do so on Monday. If it would help, we can loop in someone else, like original author(s) of the payments task?)

Noticed something that seemed to me like a new behavior from this change, though I'm not sure – a flash of incomplete styling on page load [[screenshot](https://cloudup.com/cxXl3_WcuYv)] before resolving to the proper appearance. (Maybe somehow related to https://github.com/woocommerce/woocommerce-admin/pull/4345?)

Reply:
Argh, that happened because I rebased, the problem (which most times is imperceptible) is that in https://github.com/woocommerce/woocommerce-admin/pull/4345 I only updated `recommendedMethod` on `didComponentUpdate`, and not in the constructor too. I've "reverted" that change in https://github.com/woocommerce/woocommerce-admin/pull/4499/commits/97e153f592e35a57e7b032d7122c20005866d055